src/validation/validationRules.js

```diff
@@ -1,6 +1,6 @@
 
 import moment from 'moment'
-import {includes, every} from 'lodash';
+import {includes, every, isNull} from 'lodash';
```

Comment:
Why are we adding isNull in the validators? What happens when e.g. data from the API has some null fields, it seems with this validator we are allowing null fields to go back into the API?

Especially since now we are nullifying some fields, it would seem reselecting them would allow us to post null data in new languages in the API? Or am I missing something?

Reply:
> Why are we adding isNull in the validators? What happens when e.g. data from the API has some null fields, it seems with this validator we are allowing null fields to go back into the API?

The validation of the form is now done after the values for deselected languages have been nullified. Those nullified values will be passed to the validator, hence the added null check. This was done to fix the issue where invalid values for deselected languages triggered validation warnings.

**Example:**
1. fi and sv are selected as the languages.
2. The user enters a _valid_ value for fi and an _invalid_ value for sv to the homepage url field.
3. The user then unselects sv, so that fi remains the only selected language.
4. The user saves the (seemingly valid) form. (Let's assume that the rest of the fields are valid)

This would result in a validation popup being shown for the homepage url, as the unselected language values hadn’t been cleared yet. The values for languages are _not_ cleared when a user un- or reselects a language, it's done only when the save button is pressed.

> Especially since now we are nullifying some fields, it would seem reselecting them would allow us to post null data in new languages in the API? Or am I missing something?

I'm not quite sure I understand.

Reply:
Okay, I believe this is OK in the current API implementation. Because this will result in `null` values being posted to the API, as opposed to missing dictionary keys.

The current API spec doesn't tell what should happen if the UI posts `null` in a specific language field. Luckily, the internal implementation is such that `null` language fields are not returned by the API in GET, so we are fine.

Earlier, we had issues with the API actually returning `null` for missing languages, if `null` was POSTed for a specific language. Currently, it should leave them out, though.

Example:

1) the UI POST contains an object 
```
{
`fi`: `text`,
`sv`: null
}
```

2) GETting the same resource will return an object
```
{
`fi`: `text`
}
```

with the null field missing. So this should be fine.

Reply:
So, my question was just whether the UI should POST any nullified languages at all, or not. But most likely the results are identical in both cases.

Reply:
So, if the UI is fine with both missing language keys and null valued language keys (and behaves identically in both cases), this is fine.